### PR TITLE
Fix null "branch" field for generated courses

### PR DESCRIPTION
**Pull request by nicknytko**
Fixes `null` branches for newly generated courses by looking up the branch that was created on GitHub

### CODE REVIEW

**Comment by nicknytko** [comment]
still needs to be tested before i'd feel comfortable merging this

File: lib/github.js [hunk 1]
```diff
@@ -254,6 +254,18 @@ module.exports = {
...
                 job.info(`Created repository ${options.repo_short_name}`);
+
+                /* Find main branch (which is the only branch in the new repo), output of this is array of objects following:
+                   https://docs.github.com/en/rest/reference/repos#list-branches */
+                const branches = await client.repos.listBranches({
+                    owner: config.githubCourseOwner,
+                    repo: options.repo_short_name,
+                });
+                if (branches.length != 1) {
+                    throw `New repo has ${branches.length} branches, expected one.`;
```
Comment (mwest1066) [inline lib/github.js:265]: https://stackoverflow.com/a/11512623
```suggestion
                    throw new Error(`New repo has ${branches.length} branches, expected one.`);
```

**Review by mwest1066 — APPROVED** [review]

**Comment by mwest1066** [comment]
> still needs to be tested before i'd feel comfortable merging this

Do you have things set up to test it locally, or should we just merge it?

**Comment by nicknytko** [comment]
> > still needs to be tested before i'd feel comfortable merging this
>
> Do you have things set up to test it locally, or should we just merge it?

I should be able to copy all the auth tokens and config settings from prod to get this working locally. I’ll give this a try today.

**Comment by mwest1066** [comment]
> I should be able to copy all the auth tokens and config settings from prod to get this working locally. I’ll give this a try today.

Excellent, poke me when it's ready to merge.

**Comment by nicknytko** [comment]
it's now getting the branch correctly:

```
Inserted course into database:
{
    "id": "1",
    "short_name": "TEST 105",
    "title": "branch test",
    "display_timezone": "America/Chicago",
    "grading_queue": null,
    "path": "/data1/courses/pl-test105",
    "repository": "git@github.com:PrairieLearn/pl-test105.git",
    "deleted_at": null,
    "options": {},
    "commit_hash": null,
    "institution_id": "1",
    "example_course": false,
    "sync_job_sequence_id": null,
    "sync_errors": null,
    "sync_warnings": null,
    "branch": "master",
    "manual_grading_visible": false
}
```

**Comment by mwest1066** [comment]
> ```
>     "branch": "master",
> ```

Awesome!

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**